crates/composition/src/diagnostics.rs

```diff
@@ -0,0 +1,28 @@
+/// Warnings and errors produced by composition.
+#[derive(Default)]
+pub struct Diagnostics(Vec<Diagnostic>);
```

Comment:
This module is a very very rough sketch, we'll want more info in our diagnostics.

Reply:
Also in general error handling: there are few errors to handle, but still we shouldn't panic as much as in this sketch.